annotations/views/project_views.py

```diff
@@ -179,3 +185,67 @@ def list_projects(request):
         'projects': qs,
     }
     return render(request, template, context)
+
+
+
+def upload(request):
+
+    context = {}
+    pathForm = PathForm()
+    
+    context = {}
+    if request.method == 'POST':
+        pathForm = PathForm(request.POST)
+        if pathForm.is_valid():
+            csv_file = request.FILES['csv_file']
+            decoded_file = csv_file.read().decode('utf-8')
+            io_string = io.StringIO(decoded_file)
+            csvreader = csv.reader(io_string, delimiter=',', quotechar='"')
+            csvreader.next()
+            project = request.session.get('project', 'none')
+            for row in csvreader:
+                try:
+                    parent = Text.objects.get(uri=row[3])
+                    text = Text.objects.get(part_of_id=parent.id)
+                    occur = text
+                except:
+                    url = "https://amphora.asu.edu/amphora/resource/get?uri=" + row[3] + "&format=json"
+                    r = requests.get(url, headers=auth.jars_github_auth(request.user))
+                    text_json = r.json()
+                    found = False
+                    while found == False:
+                        for content in text_json['content']:
+                            if content['content_resource']['content_type'] == 'text/plain':
+                                text_content = content['content_resource']['id']
+                                found = True
+                    repository_text_add_to_project(request, 1, text_json['id'],project)
+                    text = Text.objects.get(uri=row[3])
+                    repository_text_content(request, 1, text_json['id'], text_content)
```

Comment:
don't call a view here. especially since you're not using the returned http object. If the view does something that you need, put that something into it's own method and call it from both views.